source/extensions/filters/http/oauth2/filter.cc

```diff
@@ -145,9 +145,13 @@ std::string encodeHmac(const std::vector<uint8_t>& secret, absl::string_view hos
   auto& crypto_util = Envoy::Common::Crypto::UtilitySingleton::get();
   const auto hmac_payload =
       absl::StrJoin({host, expires, token, id_token, refresh_token}, HmacPayloadSeparator);
-  std::string encoded_hmac;
-  absl::Base64Escape(Hex::encode(crypto_util.getSha256Hmac(secret, hmac_payload)), &encoded_hmac);
-  return encoded_hmac;
+  if (Runtime::runtimeFeatureEnabled("envoy.reloadable_features.hmac_hex_encoding_only")) {
+    return Hex::encode(crypto_util.getSha256Hmac(secret, hmac_payload));
```

Comment:
Hex will take two characters for each byte while Base64 takes 4 characters for every 3 bytes, so it should be more efficient to just use base64. Wdyt?

Reply:
I agree, we can go with Base64.

Reply:
Do you have any guidance on 'respecting the token of the old format'? 
I can update the OAuth2 filter validator to accept both 'base64' and 'hex+base64'. However, how do we disable this behavior when the user no longer needs this backward compatibility?

Reply:
I agree accepting the old token is a good idea. What we may be able to do is to accept the base64+hex token and `set-cookie` with the new token. This way when this backward compatibility is not required and becomes the default behaviour we can remove the runtime guard safely. There may be better ideas, but this seems simple enough.